### DISCUSSION TITLE
A brave new interface

### DISCUSSION THREAD

**Discussion by sailxjx** [opendilab/DI-engine#84] [General]
We will design a brave new interactive interface in Di-engine 1.0, including program api and cli commands, which will support most reinforcement learning scenarios, and the rest will be implemented by our elastic `atomic` components.

Here are some design guidelines for the new interfaces:

1. It should be compatible with the existing configurations, and easy to convert from the old code to the new style.
2. It should be semantic and easy to understand, anyone with little reinforment learning experience will benifit from the policy examples.
3. It should be easy to extend to multi-threaded or distributed environments.

Any suggestions are welcome, please leave your comments in this channel.

**Comment by zxzzz0** [comment]
> by our elastic `atomic` components.  

May us know what is this?

**Reply by sailxjx** [reply]
It's still in plan, you can think of it as a layer in a neural network, but for RL policies. However, because a policy has more possibilities and less structured than a model, the abstraction of basic elements of a policy can be very challenging...
